web/concrete/controllers/single_page/dashboard/system/seo/urls.php

```diff
@@ -1,144 +1,173 @@
-    * @return void
-    */
-    public function view($strStatus = false, $blnHtu = false)
-    {
         $strStatus = (string) $strStatus;
-        $blnHtu = (bool) $blnHtu;
-        $intRewriting = Config::get('concrete.seo.url_rewriting') == 1 ? 1 : 0;
-
-        $this->set('fh', \Core::make('helper/form'));
-        $this->set('strRules', $this->getRewriteRules());
-        $this->set('intRewriting', $intRewriting);
-        $this->set('canonical_url', Config::get('concrete.seo.canonical_url'));
-        $this->set('canonical_ssl_url', Config::get('concrete.seo.canonical_ssl_url'));
-        $this->set('redirect_to_canonical_url', Config::get('concrete.seo.redirect_to_canonical_url'));
-
-        if ($strStatus == 'saved') {
+        if ($strStatus === 'saved') {
             $message = t('Settings Saved.');
-            if (Config::get('concrete.seo.url_rewriting') && !$blnHtu) {
-                $this->set('message', $message . ' ' . $urlmsg . ' ' . t('You need to update .htaccess by hand.'));
-            } elseif (Config::get('concrete.seo.url_rewriting') && $blnHtu) {
-                $this->set('message', $message . ' ' . $urlmsg . ' ' .t('We were able to automatically update .htaccess file.'));
-            } else {
-                $this->set('message', $message);
+            $prettyUrlState = (string) $prettyUrlState;
+            switch ($prettyUrlState) {
+                case 'saved':
+                case 'not-needed':
+                    $manager = $this->app->make('Concrete\Core\Service\Manager\ServiceManager');
+                    $services = $manager->getActiveServices();
+                    $service = $services[0];
+                    $rule = $service->getGenerator()->getRule('pretty_urls');
+                    switch ($prettyUrlState) {
+                        case 'saved':
+                            if ($urlRewriting) {
+                                $this->set('configuration_action', t('The following rule has been added to the server configuration'));
+                            } else {
+                                $this->set('configuration_action', t('The following rule has been removed to the server configuration'));
+                            }
+                            break;
+                        case 'not-needed':
+                            if ($urlRewriting) {
+                                $this->set('configuration_action', t('The following rule was already in your server configuration'));
+                            } else {
+                                $this->set('configuration_action', t('The following rule was already missing in your server configuration'));
+                            }
+                            break;
+                    }
+                    $this->set('configuration_code', $rule['code']);
+                    break;
+                case 'unrecognized':
+                    $codes = array();
```

Comment:
I'd probably put the different cases here in private methods. This would make it easier to read all separate blocks.